### PR TITLE
chore: remove step for authenticated master session check to use standard script

### PR DESCRIPTION
**Pull request by amandavialva01**
chore: remove step for authenticated master session check to use standard script

## Ticket
No ticket.

## Description
`test-e2e-gke-shared-cluster` adds a step that waits for the master connection using an authenticated determined session, verifying that valid authenticated requests during the test runs are expected to succeed. 
Remove this step to use the standard script that tests the master connection with an unauth session, as this achieves the main goal of the step, which is to ensure that the master is accessible prior to running test cases.

## Test Plan
CI passes.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by JComins000 — APPROVED** [review]

File: .circleci/real_config.yml [hunk 1]
```diff
@@ -3774,24 +3774,24 @@ jobs:
...
-          name: Wait for master connection
-          command: |
-            set +o pipefail
-            export DET_USER=admin DET_PASS=${INITIAL_USER_PASSWORD}
-            export DET_MASTER_TLS_CERT=${MASTER_TLS_CERT} DET_MASTER_CERT_NAME=${MASTER_CERT_NAME}
-            for i in {1..10}; do
-              yes | det -m https://${MASTER_HOST}:${MASTER_PORT} user whoami | grep "logged in" && break || \
-              echo "Trying to connect to master host again in 5 seconds" && sleep 5
-            done
+      # - run:
```
Comment (JComins000) [inline .circleci/real_config.yml:3777]: can we delete this?

**Review by JComins000 — CHANGES_REQUESTED** [review]
one comment. also, why doesn't this issue have a ticket?

File: .circleci/real_config.yml (hunk 1 above)
Comment (amandavialva01) [reply .circleci/real_config.yml]: Yep! Was just checking that the tests would still pass, removed!

**Review by JComins000 — APPROVED** [review]

**Comment by amandavialva01** [comment]
> one comment. also, why doesn't this issue have a ticket?

This was inquired about in a [Slack thread](https://hpe-aiatscale.slack.com/archives/C04C9JXB1C2/p1715189395280589). Since it's a quick change, I just made a PR!

**Pull request merged**